front/src/Api/IframeListener.ts

```diff
@@ -106,14 +105,22 @@ class IframeListener {
     private readonly _setTilesStream: Subject<SetTilesEvent> = new Subject();
     public readonly setTilesStream = this._setTilesStream.asObservable();
 
+    private readonly _setVariableStream: Subject<SetVariableEvent> = new Subject();
+    public readonly setVariableStream = this._setVariableStream.asObservable();
+
     private readonly iframes = new Set<HTMLIFrameElement>();
     private readonly iframeCloseCallbacks = new Map<HTMLIFrameElement, (() => void)[]>();
     private readonly scripts = new Map<string, HTMLIFrameElement>();
     private sendPlayerMove: boolean = false;
 
-    private answerers: {
-        [key in keyof IframeQueryMap]?: AnswererCallback<key>
-    } = {};
+
+    // Note: we are forced to type this in "any" because of https://github.com/microsoft/TypeScript/issues/31904
```

Comment:
you dont really need to type it as any 
instead you can also https://github.com/thecodingmachine/workadventure/blob/9bcdc9ba338df6fcfe6140e217d036bac18ea950/front/src/Api/IframeListener.ts#L162 force type at usages (to unknown if neccessary ) that way static accesses like this.answererers["getState"] are typed correctly

Reply:
Excellent idea! :+1: